src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationapi/timed/RemoveAccessTokens.kt

```diff
@@ -12,31 +12,47 @@ import uk.gov.justice.digital.hmpps.authorizationapi.data.repository.Authorizati
 class RemoveAccessTokens(private val service: RemoveAccessTokensService, private val telemetryClient: TelemetryClient) {
 
   @Scheduled(cron = "\${application.authentication.cc-token.cron}", zone = "Europe/London")
-  fun removeClientCredentialsAccessToken() {
+  fun removeAllButLatestAccessToken() {
     try {
-      service.removeAccessTokens()
-      log.trace("Authorization API client credentials access-tokens removed")
+      val numberOfRecordsDeleted = service.removeAllButLatestAccessToken()
 
-      telemetryClient.trackEvent("AuthorizationApiCCAccessTokensRemoved", null, null)
+      log.trace("Authorization API client,{} all but latest access tokens are removed", numberOfRecordsDeleted)
+
+      telemetryClient.trackEvent("AuthorizationApiAccessTokensRemoved", mapOf("numberOfRecordsDeleted" to numberOfRecordsDeleted.toString()), null)
     } catch (e: Exception) {
       // have to catch the exception here otherwise scheduling will stop
-      log.error("Caught exception {} during access token removal", e.javaClass.simpleName, e)
+      log.error("Caught exception {} during removal of all but latest access tokens", e.javaClass.simpleName, e)
     }
   }
 
   @Scheduled(cron = "\${application.authentication.ac-token.cron}", zone = "Europe/London")
-  fun removeClientAuthorizationCodeAccessToken() {
+  fun removeAllAuthorizationCodeRecordsWithoutAccessTokens() {
     try {
-      service.removeAuthCodeAccessTokens()
-      log.trace("Authorization API authorization-code access-tokens removed")
+      val numberOfRecordsDeleted = service.removeAuthCodeAccessTokens()
 
-      telemetryClient.trackEvent("AuthorizationApiACAccessTokensRemoved", null, null)
+      log.trace("Authorization API,{} records removed without access tokens", numberOfRecordsDeleted)
```

Comment:
Change trace message to
log.trace("Authorization API removed {} expired authorisation code records without access tokens", numberOfRecordsDeleted)